### DISCUSSION TITLE
Upgrade glances to newer version

### DISCUSSION THREAD

**Discussion by LarryT8959** [nicolargo/glances#2818] [General]
I have multiple instances of glances running all over my lab in vm's, bare metal on Mac, and in containers. What is the best way to upgrade? Best command line to use in these situations without loosing settings?

**Comment by nicolargo** [comment]
There is no "magics" tricks. My advice is to backup yours Glances configuration files, update Glances (method depend on the way it has been installed) and merge your previous configuration file with the new one.

If you upgrade from Glances v3 to Glances v4, be aware that there are some [breaking changes](https://github.com/nicolargo/glances/wiki/Glances-4.0-Release-Note#breaking-changes).

**Comment by LarryT8959** [comment]
I'm pretty much back to where I was, but this wasn't an easy fix.  I couldn't get one of my pi's to work with widgets, the other one is fine. The only reason I can think of is because it's the older PI?? I think it's a 3, or 2 if it's such a thing?. I basically started from scratch and reinstalled all software. The VM's weren't that bad, I basically had to update everything.
